server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/AttachmentCheckerTest.java

```diff
@@ -0,0 +1,105 @@
+    private final CreationMessage.Builder creationMessageBuilder = CreationMessage.builder()
+        .from(CreationMessage.DraftEmailer.builder().name("alice").email("alice@example.com").build())
+        .to(ImmutableList.of(CreationMessage.DraftEmailer.builder().name("bob").email("bob@example.com").build()))
+        .mailboxId("id")
+        .subject("Hey! ");
+
+    private AttachmentManager attachmentManager;
+    private MailboxSession session;
+
+    private AttachmentChecker sut;
+
+    @Before
+    public void setUp() {
+        session = new MockMailboxSession("Jonhy");
+        attachmentManager = mock(AttachmentManager.class);
+
+        sut = new AttachmentChecker(attachmentManager);
+    }
+
+    @Test
+    public void assertAttachmentsExistShouldThrowWhenUnknownBlobId() throws MailboxException {
+        BlobId unknownBlobId = BlobId.of("unknownBlobId");
+        AttachmentId unknownAttachmentId = AttachmentId.from(unknownBlobId.getRawValue());
+        when(attachmentManager.getAttachment(unknownAttachmentId, session)).thenThrow(new AttachmentNotFoundException(unknownBlobId.getRawValue()));
+
+        assertThatThrownBy(() -> sut.assertAttachmentsExist(
+            new ValueWithId.CreationMessageEntry(
+                creationMessageId,
+                creationMessageBuilder.attachments(
+                    Attachment.builder().size(12L).type("image/jpeg").blobId(unknownBlobId).build())
+                    .build()
+            ),
+            session))
+            .isInstanceOf(AttachmentsNotFoundException.class);
+    }
+
+    @Test
+    public void assertAttachmentsExistShouldThrowWhenUnknownBlobIds() throws MailboxException {
+        BlobId unknownBlobId1 = BlobId.of("unknownBlobId1");
+        BlobId unknownBlobId2 = BlobId.of("unknownBlobId2");
+        AttachmentId unknownAttachmentId1 = AttachmentId.from(unknownBlobId1.getRawValue());
+        AttachmentId unknownAttachmentId2 = AttachmentId.from(unknownBlobId2.getRawValue());
+
+        when(attachmentManager.getAttachment(unknownAttachmentId1, session)).thenThrow(new AttachmentNotFoundException(unknownBlobId1.getRawValue()));
+        when(attachmentManager.getAttachment(unknownAttachmentId2, session)).thenThrow(new AttachmentNotFoundException(unknownBlobId2.getRawValue()));
+
+        assertThatThrownBy(() -> sut.assertAttachmentsExist(
+            new ValueWithId.CreationMessageEntry(
+                creationMessageId,
+                creationMessageBuilder.attachments(
+                    Attachment.builder().size(12L).type("image/jpeg").blobId(unknownBlobId1).build(),
+                    Attachment.builder().size(23L).type("image/git").blobId(unknownBlobId2).build())
+                    .build()
+            ),
+            session))
+            .isInstanceOf(AttachmentsNotFoundException.class)
+            .matches(e -> ((AttachmentsNotFoundException)e).getAttachmentIds().containsAll(ImmutableSet.of(unknownBlobId1, unknownBlobId2)));
+    }
```

Comment:
you have 2 negative test cases, can you add positive test cases?
assertAttachmentExistsShouldNotThrowWhenKnownBlobId
assertAttachmentsExistShouldNotThrowWhenKnownBlobIds

Reply:
We would also need the mix between positive and negative.

Ps: I did not do it as I only intended to move code around.